src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.EditAndContinue
+{
+    /// <summary>
+    /// Retrieves the <see cref="Solution"/> snapshot that corresponds to the current state of the debuggee.
+    /// This snapshot contains changes successfully applied during EnC/Hot Reload.
+    /// </summary>
+    /// <remarks>
+    /// This is temporarily available to in-proc XAML External Access APIs and should be revisited once XAML moves to OOP LSP.
+    /// </remarks>
+    internal interface IEditAndContinueSolutionProvider
```

Comment:
>internal [](http://example.com/codeflow?start=4&length=8)

This should be public

Reply:
If this is only temporary, then you don't want to make this public and therefore available to anyone. You can consume this from our XAML assembly in Roslyn and expose another internal interface from there that we can see from XamlLanguageService in VS via IVT.

Reply:
OK I didn't know about a XAML assembly in Roslyn. That works as long as all the internals are visible to the right assemblies.

Reply:
It's only visible to one assembly, Microsoft.VisualStudio.DesignTools.XamlLanguageService, but that should be enough.